test/ConfigAggregatorTest.php

```diff
@@ -57,6 +57,22 @@ protected function tearDown(): void
         @rmdir(dirname($this->cacheFile));
     }
 
+    public function testConfigAggregatorRaisesExceptionOnDuplicateProvider(): void
+    {
+        $configProvider = new class
```

Comment:
Wait. This works? More instances could be created for anonymous class using generated class name?

Reply:
yup